tooling/backend_interface/src/lib.rs

```diff
@@ -104,6 +107,18 @@ impl Backend {
     fn crs_directory(&self) -> PathBuf {
         self.backend_directory().join("crs")
     }
+
+    fn assert_correct_version(&self) {
+        let binary_path = self.binary_path();
+        if binary_path.to_string_lossy().contains(BACKEND_BARRETENBERG_SEARCH_STR) {
+            let version_result = VersionCommand.run(binary_path);
+            if let Ok(version_string) = version_result {
+                if version_string.as_str() != BB_VERSION {
+                    println!("WARNING!: Configured backend version `{version_string}` is different from expected `{BB_VERSION}`");
```

Comment:
```suggestion
                // If we're trying to use barretenberg, automatically go and install the correct version.
                let bb_url = std::env::var("BB_BINARY_URL")
                    .unwrap_or_else(|_| bb_abstraction_leaks::BB_DOWNLOAD_URL.to_owned());
                download_backend(&bb_url, binary_path)?;
```

The function will then need to return a `Result`, alternatively we can move this check to be an implementation detail inside `assert_binary_exists`.

Reply:
(We'll also need to delete the existing binary)

Reply:
What if it's desired state? Seeing a warning is prompt to react but if this state is deliberate, it can safely be ignored.

Reply:
> What if it's desired state?

Can you describe when having `acvm-backend-barretenberg` being the incorrect version be the desired state for an end user?

The vast majority of the time I don't think printing this warning will help users as they won't know how to resolve this issue without asking in the discord (which they're doing already).

Reply:
Then we can fix the message, instructing to delete backend at particular location, I agree with this...
Still a developer may want to test different versions of a backend and go about it as they see fit.

Reply:
Automatically fixing `acvm-backend-barretenberg` doesn't prevent a developer from doing this though. A developer can install arbitrary backends under different names and test as they see fit. I'm not clear on the benefit of allowing this to be done under the name `acvm-backend-barretenberg` however.

I think that we can provide the UX that the default backend `acvm-backend-barretenberg` "just works", if you want to do anything more fancy then you can do it as a custom backend but we can't give any guarantees around that.

Reply:
4be4041: chore: replace-download correct backend version
is a little forcefull but I like ease of use more.